evap/staff/tests/test_views.py

```diff
@@ -3090,6 +3090,92 @@ def test_archived(self):
         self.assert_transition("publish", TextAnswer.ReviewDecision.UNDECIDED, status=403)
 
 
+class TestEvaluationTextanswersUpdateFlagView(WebTest):
+    url = reverse("staff:evaluation_textanswers_update_flag")
+    csrf_checks = False
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.manager = make_manager()
+        cls.answer = baker.make(TextAnswer, contribution__evaluation__can_publish_text_results=True)
+
+    def test_post_update(self):
+        with run_in_staff_mode(self):
+            self.assertFalse(self.answer.is_flagged)
+
+            # Add flag
+            self.app.post(
+                self.url,
+                user=self.manager,
+                status=204,
+                params={"answer_id": self.answer.pk, "is_flagged": "true"},
+            )
+            self.answer.refresh_from_db()
+            self.assertTrue(self.answer.is_flagged)
+
+            # Do it again
+            self.app.post(
+                self.url,
+                user=self.manager,
+                status=204,
+                params={"answer_id": self.answer.pk, "is_flagged": "true"},
+            )
+            self.answer.refresh_from_db()
+            self.assertTrue(self.answer.is_flagged)
+
+            # Remove flag
+            self.app.post(
+                self.url,
+                user=self.manager,
+                status=204,
+                params={"answer_id": self.answer.pk, "is_flagged": "false"},
+            )
+            self.answer.refresh_from_db()
+            self.assertFalse(self.answer.is_flagged)
+
+    def test_unknown_values(self):
+        with run_in_staff_mode(self):
+            self.app.post(self.url, user=self.manager, status=400, params={"answer_id": self.answer.pk})
+            self.answer.refresh_from_db()
+            self.assertFalse(self.answer.is_flagged)
+
+            self.app.post(
+                self.url,
+                user=self.manager,
+                status=400,
+                params={"answer_id": self.answer.pk, "is_flagged": "True"},
+            )
+            self.answer.refresh_from_db()
+            self.assertFalse(self.answer.is_flagged)
```

Comment:
The blocks here could be a loop of `[("True", False), ("False", False), ("", False), ("true", True)]`, which I think I'd prefer to reduce duplication.